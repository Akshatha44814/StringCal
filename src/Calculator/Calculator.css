.conatiner {
  background-color: white;
  padding: 2%;
  margin: 2%;
  width: 500px;
  border-radius: 2%;
  box-shadow: 2px 2px 5px lightgray;
}
.calculate {
  background: blue;
  color: white;
  padding: 10px;
  width: 385px;
  border: 1px solid #ececec;
  border-radius: 4px;
  margin: 10px;
}
.colorContent {
  color: red;
}
.content {
  display: flex;
  margin: 2% 10% 2% 12%;
}
