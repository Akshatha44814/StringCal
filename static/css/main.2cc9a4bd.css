body{-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;align-items:center;background-color:#f0f8ff;display:flex;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue,sans-serif;height:650px;justify-content:center;margin:0;text-align:center}code{font-family:source-code-pro,Menlo,Monaco,Consolas,Courier New,monospace}.App{text-align:center}.App-logo{height:40vmin;pointer-events:none}@media (prefers-reduced-motion:no-preference){.App-logo{animation:App-logo-spin 20s linear infinite}}.App-header{align-items:center;background-color:#282c34;color:#fff;display:flex;flex-direction:column;font-size:calc(10px + 2vmin);justify-content:center;min-height:100vh}.App-link{color:#61dafb}@keyframes App-logo-spin{0%{transform:rotate(0deg)}to{transform:rotate(1turn)}}.input{border:1px solid #ececec;border-radius:2px;padding:10px;width:360px}.resultInput{margin-bottom:5%}button[disabled]{background-color:#ccc;border:1px solid #999;color:#666}.conatiner{background-color:#fff;border-radius:2%;box-shadow:2px 2px 5px #d3d3d3;margin:2%;padding:2%;width:500px}.calculate{background:blue;border:1px solid #ececec;border-radius:4px;color:#fff;margin:10px;padding:10px;width:385px}.colorContent{color:red}.content{display:flex;margin:2% 10% 2% 12%}
/*# sourceMappingURL=main.2cc9a4bd.css.map*/