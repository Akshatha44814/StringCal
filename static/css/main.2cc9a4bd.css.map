{"version":3,"file":"static/css/main.2cc9a4bd.css","mappings":"AAAA,KAKE,kCAAmC,CACnC,iCAAkC,CAIlC,kBAAmB,CAHnB,wBAA2B,CAE3B,YAAa,CAPb,mIAEY,CAQZ,YAAa,CADb,sBAAuB,CAVvB,QAAS,CAOT,iBAKF,CAEA,KACE,uEAEF,CClBA,KACE,iBACF,CAEA,UACE,aAAc,CACd,mBACF,CAEA,8CACE,UACE,2CACF,CACF,CAEA,YAKE,kBAAmB,CAJnB,wBAAyB,CAOzB,UAAY,CALZ,YAAa,CACb,qBAAsB,CAGtB,4BAA6B,CAD7B,sBAAuB,CAJvB,gBAOF,CAEA,UACE,aACF,CAEA,yBACE,GACE,sBACF,CACA,GACE,uBACF,CACF,CCrCA,OAGE,wBAAyB,CACzB,iBAAkB,CAHlB,YAAa,CACb,WAGF,CACA,aACE,gBACF,CACA,iBAEE,qBAAyB,CADzB,qBAAyB,CAEzB,UACF,CCbA,WACE,qBAAuB,CAIvB,gBAAiB,CACjB,8BAAiC,CAHjC,SAAU,CADV,UAAW,CAEX,WAGF,CACA,WACE,eAAgB,CAIhB,wBAAyB,CACzB,iBAAkB,CAJlB,UAAY,CAKZ,WAAY,CAJZ,YAAa,CACb,WAIF,CACA,cACE,SACF,CACA,SACE,YAAa,CACb,oBACF","sources":["index.css","App.css","CalculatorPage/CalculatorPage.css","Calculator/Calculator.css"],"sourcesContent":["body {\r\n  margin: 0;\r\n  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',\r\n    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',\r\n    sans-serif;\r\n  -webkit-font-smoothing: antialiased;\r\n  -moz-osx-font-smoothing: grayscale;\r\n  background-color: aliceblue;\r\n  text-align: center;\r\n  display: flex;\r\n  align-items: center;\r\n  justify-content: center;\r\n  height: 650px;\r\n}\r\n\r\ncode {\r\n  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',\r\n    monospace;\r\n}\r\n",".App {\r\n  text-align: center;\r\n}\r\n\r\n.App-logo {\r\n  height: 40vmin;\r\n  pointer-events: none;\r\n}\r\n\r\n@media (prefers-reduced-motion: no-preference) {\r\n  .App-logo {\r\n    animation: App-logo-spin infinite 20s linear;\r\n  }\r\n}\r\n\r\n.App-header {\r\n  background-color: #282c34;\r\n  min-height: 100vh;\r\n  display: flex;\r\n  flex-direction: column;\r\n  align-items: center;\r\n  justify-content: center;\r\n  font-size: calc(10px + 2vmin);\r\n  color: white;\r\n}\r\n\r\n.App-link {\r\n  color: #61dafb;\r\n}\r\n\r\n@keyframes App-logo-spin {\r\n  from {\r\n    transform: rotate(0deg);\r\n  }\r\n  to {\r\n    transform: rotate(360deg);\r\n  }\r\n}\r\n",".input {\r\n  padding: 10px;\r\n  width: 360px;\r\n  border: 1px solid #ececec;\r\n  border-radius: 2px;\r\n}\r\n.resultInput {\r\n  margin-bottom: 5%;\r\n}\r\nbutton[disabled] {\r\n  border: 1px solid #999999;\r\n  background-color: #cccccc;\r\n  color: #666666;\r\n}\r\n.colorContent {\r\n  color: red;\r\n}\r\n",".conatiner {\r\n  background-color: white;\r\n  padding: 2%;\r\n  margin: 2%;\r\n  width: 500px;\r\n  border-radius: 2%;\r\n  box-shadow: 2px 2px 5px lightgray;\r\n}\r\n.calculate {\r\n  background: blue;\r\n  color: white;\r\n  padding: 10px;\r\n  width: 385px;\r\n  border: 1px solid #ececec;\r\n  border-radius: 4px;\r\n  margin: 10px;\r\n}\r\n.colorContent {\r\n  color: red;\r\n}\r\n.content {\r\n  display: flex;\r\n  margin: 2% 10% 2% 12%;\r\n}\r\n"],"names":[],"sourceRoot":""}