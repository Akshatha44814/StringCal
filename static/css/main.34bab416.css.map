{"version":3,"file":"static/css/main.34bab416.css","mappings":"AAAA,KAKE,kCAAmC,CACnC,iCAAkC,CAIlC,kBAAmB,CAHnB,wBAA2B,CAE3B,YAAa,CAPb,mIAEY,CAQZ,YAAa,CADb,sBAAuB,CAVvB,QAAS,CAOT,iBAKF,CAEA,KACE,uEAEF,CClBA,KACE,iBACF,CAEA,UACE,aAAc,CACd,mBACF,CAEA,8CACE,UACE,2CACF,CACF,CAEA,YAKE,kBAAmB,CAJnB,wBAAyB,CAOzB,UAAY,CALZ,YAAa,CACb,qBAAsB,CAGtB,4BAA6B,CAD7B,sBAAuB,CAJvB,gBAOF,CAEA,UACE,aACF,CAEA,yBACE,GACE,sBACF,CACA,GACE,uBACF,CACF,CCrCA,OAGE,wBAAyB,CACzB,iBAAkB,CAClB,WAAmB,CAJnB,YAAa,CACb,SAIF,CACA,aACE,gBACF,CACA,iBAEE,qBAAyB,CADzB,qBAAyB,CAEzB,UACF,CAIA,WACE,eAAgB,CAEhB,wBAAyB,CACzB,iBAAkB,CAFlB,UAAY,CAGZ,YAAa,CACb,SACF,CAEA,yBACE,WACE,UACF,CACF,CC/BA,WACE,qBAAuB,CAIvB,gBAAiB,CACjB,8BAAiC,CAHjC,SAAU,CADV,UAAW,CAEX,WAGF,CACA,cACE,SACF,CACA,SACE,YAAa,CACb,SACF,CAEA,yBACE,WACE,WACF,CACA,SACE,QACF,CACF","sources":["index.css","App.css","CalculatorPage/CalculatorPage.css","Calculator/Calculator.css"],"sourcesContent":["body {\r\n  margin: 0;\r\n  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',\r\n    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',\r\n    sans-serif;\r\n  -webkit-font-smoothing: antialiased;\r\n  -moz-osx-font-smoothing: grayscale;\r\n  background-color: aliceblue;\r\n  text-align: center;\r\n  display: flex;\r\n  align-items: center;\r\n  justify-content: center;\r\n  height: 650px;\r\n}\r\n\r\ncode {\r\n  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',\r\n    monospace;\r\n}\r\n",".App {\r\n  text-align: center;\r\n}\r\n\r\n.App-logo {\r\n  height: 40vmin;\r\n  pointer-events: none;\r\n}\r\n\r\n@media (prefers-reduced-motion: no-preference) {\r\n  .App-logo {\r\n    animation: App-logo-spin infinite 20s linear;\r\n  }\r\n}\r\n\r\n.App-header {\r\n  background-color: #282c34;\r\n  min-height: 100vh;\r\n  display: flex;\r\n  flex-direction: column;\r\n  align-items: center;\r\n  justify-content: center;\r\n  font-size: calc(10px + 2vmin);\r\n  color: white;\r\n}\r\n\r\n.App-link {\r\n  color: #61dafb;\r\n}\r\n\r\n@keyframes App-logo-spin {\r\n  from {\r\n    transform: rotate(0deg);\r\n  }\r\n  to {\r\n    transform: rotate(360deg);\r\n  }\r\n}\r\n",".input {\r\n  padding: 10px;\r\n  width: 90%;\r\n  border: 1px solid #ececec;\r\n  border-radius: 2px;\r\n  margin: 2% 0% 2% 0%;\r\n}\r\n.resultInput {\r\n  margin-bottom: 5%;\r\n}\r\nbutton[disabled] {\r\n  border: 1px solid #999999;\r\n  background-color: #cccccc;\r\n  color: #666666;\r\n}\r\n.colorContent {\r\n  color: red;\r\n}\r\n.calculate {\r\n  background: blue;\r\n  color: white;\r\n  border: 1px solid #ececec;\r\n  border-radius: 4px;\r\n  padding: 10px;\r\n  width: 95%;\r\n}\r\n\r\n@media (max-width: 900px) {\r\n  .calculate {\r\n    width: 100%;\r\n  }\r\n}\r\n",".conatiner {\r\n  background-color: white;\r\n  padding: 2%;\r\n  margin: 2%;\r\n  width: 500px;\r\n  border-radius: 2%;\r\n  box-shadow: 2px 2px 5px lightgray;\r\n}\r\n.colorContent {\r\n  color: red;\r\n}\r\n.content {\r\n  display: flex;\r\n  margin: 3%;\r\n}\r\n\r\n@media (max-width: 900px) {\r\n  .conatiner {\r\n    width: 300px;\r\n  }\r\n  .content {\r\n    margin: 0%;\r\n  }\r\n}\r\n"],"names":[],"sourceRoot":""}