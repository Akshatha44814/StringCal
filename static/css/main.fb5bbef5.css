body{-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;align-items:center;background-color:#f0f8ff;display:flex;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue,sans-serif;height:650px;justify-content:center;margin:0;text-align:center}code{font-family:source-code-pro,Menlo,Monaco,Consolas,Courier New,monospace}.App{text-align:center}.App-logo{height:40vmin;pointer-events:none}@media (prefers-reduced-motion:no-preference){.App-logo{animation:App-logo-spin 20s linear infinite}}.App-header{align-items:center;background-color:#282c34;color:#fff;display:flex;flex-direction:column;font-size:calc(10px + 2vmin);justify-content:center;min-height:100vh}.App-link{color:#61dafb}@keyframes App-logo-spin{0%{transform:rotate(0deg)}to{transform:rotate(1turn)}}.input{border:1px solid #ececec;border-radius:2px;margin:2% 0;padding:10px;width:90%}.resultInput{margin-bottom:5%}button[disabled]{background-color:#ccc;border:1px solid #999;color:#666}.calculate{background:blue;border:1px solid #ececec;border-radius:4px;color:#fff;padding:10px;width:95%}@media (max-width:600px){.calculate{width:90%}}.conatiner{background-color:#fff;border-radius:2%;box-shadow:2px 2px 5px #d3d3d3;margin:2%;padding:2%;width:500px}.colorContent{color:red}.content{display:flex;margin:3%}@media (max-width:600px){.conatiner{width:300px}.content{margin:2%}}
/*# sourceMappingURL=main.fb5bbef5.css.map*/