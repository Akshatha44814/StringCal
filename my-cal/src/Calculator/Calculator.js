import "./Calculator.css"

const Calculator = () =>{
    return <div className="conatiner">
        <h1>String Calculator</h1>
        <div className="content">Enter numbers</div>
        <div className="content">(Format:<span className="colorContent">//[delimiter]\n[numbers]</span>)</div>
        <div><input type="text" className="input" placeholder="E.G.,//;\n1;2;3 or 1,2,3"/></div>
        <button className="calculate">Calculate</button>
    </div>
}
export default Calculator