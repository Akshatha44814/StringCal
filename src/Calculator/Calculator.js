import CalculatorPage from "../CalculatorPage/CalculatorPage";
import "./Calculator.css";

const Calculator = () => {
  return (
    <div className="conatiner">
      <h1>String Calculator</h1>
      <p className="content">Enter numbers</p>
      <p className="content">
        {"(Format:"}
        <span className="colorContent" data-testid="redColor">
          {"//[delimiter]\n[numbers]"}
        </span>
        {")"}
      </p>
      <CalculatorPage/>
    </div>
  );
};
export default Calculator;
