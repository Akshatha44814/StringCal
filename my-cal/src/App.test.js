import {
  cleanup,
  render,
  screen,
} from "@testing-library/react";
import App from "./App";
import Calculator from "./Calculator/Calculator";
import "./index.css";
import "./Calculator/Calculator.css";
afterEach(cleanup);

test("renders Main App Component", () => {
  render(<App />);
});
describe("renders Calculate Component", () => {
  it("render Calculate component", () => {
    render(<Calculator />);
  });

  it("render String Calculator header", () => {
    const { asFragment } = render(<Calculator />);
    const linkHeader = screen.getByText("String Calculator");
    expect(linkHeader).toBeInTheDocument();
    expect(asFragment()).toHaveTextContent("String Calculator");
    expect(asFragment()).toMatchSnapshot();
    expect(asFragment()).toBeTruthy();
  });

  it("render Enter numbers label", () => {
    const { asFragment } = render(<Calculator />);
    const linkLabel = screen.getByText("Enter numbers");
    expect(linkLabel).toBeInTheDocument();
    expect(asFragment()).toHaveTextContent("Enter numbers");
    expect(asFragment()).toMatchSnapshot();
    expect(asFragment()).toBeTruthy();
  });

  it("display the format label", () => {
    const Format = () => (
      <p>
        {"(Format:"}
        <span>{"//[delimiter]\n[numbers]"}</span>
        {")"}
      </p>
    );
    const { asFragment } = render(<Format />);
    expect(asFragment()).toHaveTextContent("(Format://[delimiter] [numbers])");
    expect(asFragment()).toMatchSnapshot();
    expect(asFragment()).toBeTruthy();
  });
});
