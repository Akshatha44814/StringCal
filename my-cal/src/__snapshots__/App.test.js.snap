// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String calculator  matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="conatiner"
  >
    <h1>
      String Calculator
    </h1>
    <p
      class="content"
    >
      Enter numbers
    </p>
    <p
      class="content"
    >
      (Format:
      <em
        class="colorContent"
      >
        //[delimiter]\\n[numbers]
      </em>
      )
    </p>
    <div>
      <input
        class="input"
        placeholder="E.G.,//;\\\\n1;2;3 or 1,2,3"
        type="text"
      />
    </div>
    <button
      class="calculate"
    >
      Calculate
    </button>
    <div
      class="resultInput"
    >
      <input
        class="input"
        disabled=""
        placeholder=""
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`String calculator matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="conatiner"
  >
    <h1>
      String Calculator
    </h1>
    <p
      class="content"
    >
      Enter numbers
    </p>
    <p
      class="content"
    >
      (Format:
      <em
        class="colorContent"
      >
        //[delimiter]\\n[numbers]
      </em>
      )
    </p>
    <div>
      <input
        class="input"
        placeholder="E.G.,//;\\\\n1;2;3 or 1,2,3"
        type="text"
      />
    </div>
    <button
      class="calculate"
    >
      Calculate
    </button>
    <div
      class="resultInput"
    >
      <input
        class="input"
        disabled=""
        placeholder=""
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Calculate Component 1`] = `
<DocumentFragment>
  <div
    class="conatiner"
  >
    <h1>
      String Calculator
    </h1>
    <p
      class="content"
    >
      Enter numbers
    </p>
    <p
      class="content"
      data-testid="format"
    >
      (Format:
      <em
        class="colorContent"
      >
        //[delimiter]\\n[numbers]
      </em>
      )
    </p>
    <div>
      <input
        class="input"
        placeholder="E.G.,//;\\\\n1;2;3 or 1,2,3"
        type="text"
      />
    </div>
    <button
      class="calculate"
    >
      Calculate
    </button>
    <div
      class="resultInput"
    >
      <input
        class="input"
        disabled=""
        placeholder=""
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Calculate Component display the format label 1`] = `
<DocumentFragment>
  <p>
    (Format:
    <span>
      //[delimiter]
[numbers]
    </span>
    )
  </p>
</DocumentFragment>
`;

exports[`renders Calculate Component elements of the page 1`] = `null`;

exports[`renders Calculate Component render Enter numbers label 1`] = `
<DocumentFragment>
  <div
    class="conatiner"
  >
    <h1>
      String Calculator
    </h1>
    <p
      class="content"
    >
      Enter numbers
    </p>
    <p
      class="content"
    >
      (Format:
      <span
        class="colorContent"
        data-testid="redColor"
      >
        //[delimiter]
[numbers]
      </span>
      )
    </p>
    <div
      class="conatinerPage"
    >
      <div>
        <input
          class="input"
          placeholder="E.G.,//;\\\\n1;2;3 or 1,2,3"
          type="text"
        />
      </div>
      <button
        class="calculate"
      >
        Calculate
      </button>
      <div
        class="resultInput"
      >
        <input
          class="input"
          disabled=""
          placeholder="Result Output"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Calculate Component render String Calculator header 1`] = `
<DocumentFragment>
  <div
    class="conatiner"
  >
    <h1>
      String Calculator
    </h1>
    <p
      class="content"
    >
      Enter numbers
    </p>
    <p
      class="content"
    >
      (Format:
      <span
        class="colorContent"
        data-testid="redColor"
      >
        //[delimiter]
[numbers]
      </span>
      )
    </p>
    <div
      class="conatinerPage"
    >
      <div>
        <input
          class="input"
          placeholder="E.G.,//;\\\\n1;2;3 or 1,2,3"
          type="text"
        />
      </div>
      <button
        class="calculate"
      >
        Calculate
      </button>
      <div
        class="resultInput"
      >
        <input
          class="input"
          disabled=""
          placeholder="Result Output"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Calculate Component should render the value of negative number and error message 1`] = `
<DocumentFragment>
  <span
    class="content colorContent"
  >
    Negative numbers not allowed :-1,-2
  </span>
</DocumentFragment>
`;
